cmd/network/setup_linux.go

```diff
@@ -121,21 +190,48 @@ func main() {
 	}
 	vmSwitchCmd := exec.Command(nsenter, args...)
 
-	connFile, err := vsockConn.File()
-	if err != nil {
-		logrus.Fatal(err)
-	}
 	// pass in the vsock connection as a FD to the
 	// vm-switch process in the newely created namespace
 	vmSwitchCmd.ExtraFiles = []*os.File{connFile}
-	if err := vmSwitchCmd.Start(); err != nil {
-		logrus.Fatalf("could not start the vm-switch process: %v", err)
+	return vmSwitchCmd
+}
+
+func createVethPair(defaultNsPid, peerNsPid int) error {
```

Comment:
I think it would be nicer to pass the veth names in too; up to you.

Reply:
Right, they got left behind.